tests/timeseries/test_formula_printer.py

```diff
@@ -0,0 +1,92 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the FormulaPrinter."""
+
+
+from frequenz.sdk.timeseries._formula_engine._formula_printer import FormulaPrinter
+from frequenz.sdk.timeseries._formula_engine._formula_engine import (
+    FormulaBuilder,
+    FormulaStep,
+)
+from frequenz.sdk.timeseries._quantities import Quantity
+from frequenz.channels import Broadcast
+from frequenz.sdk.timeseries import Sample
+
+from frequenz.sdk.timeseries._formula_engine._tokenizer import (
+    Tokenizer,
+    TokenType,
+)
+
+import logging
+
+
+LOGGER = logging.getLogger(__name__)
```

Comment:
I would just use `print()` in tests, otherwise you need to pass some options to `pytest` to be able to see the log output.

Reply:
@llucax but `print()` calls weren't printed during `nox` call. Do you know how I can get print() being printed when calling `nox ... pytest ...`?

Reply:
They should be printed when test fail automatically though. To print them unconditionally: `nox -e pytest_max -- -s` (or `pytest -s`). While to see the logs you need to pass `--log-cli-level=DEBUG` to see the log output even for failing tests.

Reply:
(BTW, please don't mark ongoing conversations as resolved, I missed this one, only found it because I explicitly expanded the closed conversation)

Reply:
@llucax I only mark conversations as resolved if I consider them resolved.

Reply:
Sometimes I don't see very good. :P